src/Microsoft.AspNet.Mvc.DataAnnotations/DependencyInjection/MvcDataAnnotationsMvcCoreBuilderExtensions.cs

```diff
@@ -71,6 +73,8 @@ internal static void AddDataAnnotationsServices(IServiceCollection services)
         {
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, MvcDataAnnotationsMvcOptionsSetup>());
+            var attributeAdapterProvider = new ValidationAttributeAdapterProvider();
+            services.TryAddSingleton<IValidationAttributeAdapterProvider>(s => attributeAdapterProvider);
```

Comment:
@pranavkm your suggestion isn't an overload that exists, @dougbu yours works fine and I agree it's cleaner.